addons/portal/controllers/thread.py

```diff
@@ -20,3 +21,16 @@ def _is_message_editable(self, message, **kwargs):
         if message._is_editable_in_portal(**kwargs):
             return True
         return super()._is_message_editable(message, **kwargs)
+
+    def _get_fetch_domain(self, thread_model, thread_id, **kwargs):
+        domain = super()._get_fetch_domain(thread_model, thread_id, **kwargs)
+        model = request.env[thread_model]
+        if kwargs.get("portal"):
```

Comment:
Restricting to internal subtypes for portal access should be a security requirement, it cannot come from user input, and we currently don't rely on ACL because `_message_fetch` is called as sudo for some reason (can be remove that sudo?)

Reply:
I think the internal subtype check should be moved in mail, and its presence should condition whether sudo is used or not.